scipy/stats/_stats.pyx

```diff
@@ -300,3 +300,66 @@ def _weightedrankedtau(ordered[:] x, ordered[:] y, intp_t[:] rank, weigher, bool
     tau = ((tot - (v + u - t)) - 2. * exchanges_weight[0]
            ) / np.sqrt(tot - u) / np.sqrt(tot - v)
     return min(1., max(-1., tau))
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _gaussian_kernel_estimate(double[:, :] points, double[:, :] values,
+                              double[:, :] xi, double[:, :] precision):
+    """
+    def gaussian_kernel_estimate(double[:, :] points, double[:, :] values,
+                                  double[:, :] xi, double[:, :] precision)
+
+    Evaluate a multivariate Gaussian kernel estimate.
+
+    Parameters
+    ----------
+    points : double[:, :] with shape (n, d)
+        Data points to estimate from in d dimenions.
+    values : double[:, :] with shape (n, p)
+        Multivariate values associated with the data points.
+    xi : double[:, :] with shape (m, d)
+        Coordinates to evaluate the estimate at in d dimensions.
+    precision : double[:, :] with shape (d, d)
+        Precision matrix for the Gaussian kernel.
+
+    Returns
+    -------
+    estimate : double[:, :] with shape (m, p)
+        Multivariate Gaussian kernel estimate evaluated at the input coordinates.
+    """
+    cdef:
+        double[:, :] estimate
+        int i, j, k
+        int n = points.shape[0]
+        int d = points.shape[1]
+        int p = values.shape[1]
+        int m = xi.shape[0]
+        double arg, residual, norm
+
+    assert xi.shape[1] == d, "points and xi must have same trailing dim"
+    assert precision.shape[0] == d and precision.shape[1] == d, \
+        "precision matrix must match data dims"
+
+    # Evaluate the normalisation
+    norm = np.sqrt(np.linalg.det(precision) / (2 * np.pi) ** d)
```

Comment:
Could you be reusing `self._norm_factor` here?

Reply:
I should have removed the `self._norm_factor` from the `gaussian_kde` class. The idea is that this method can serve as a generic kernel smoother without needing side inputs to enable future functionality, e.g. a Nadaraya Watson estimator with a Gaussian kernel.

Reply:
Turns out the norm factor is also used by `logpdf`, and it can't be dropped at this point. Nevertheless, I believe that having this function be independent of the implementation of `gaussian_kde` would be preferable. Thoughts?

Reply:
maybe you could use np.prod(np.diag(whitening))**2 here instead of recomputing the determinant, should be faster?